### ISSUE TITLE
 clingo error on fedora 37 (pre-beta) with python-3.11.0rc1

### ISSUE THREAD

**Issue by balay** [LLNL/spack#32559]
### Steps to reproduce

```console
[balay@pj01 spack.y]$ ./bin/spack spec zlib
```

### Error message

```console
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Bootstrapping clingo from pre-built binaries
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Bootstrapping clingo from sources
[+] /usr (external bison-3.8.2-vlaxtemuex5o7cubhbnu4js7qrxxbz3t)
[+] /usr (external cmake-3.24.1-iitagh2olyrd3rmgviph76tqlzlc42ix)
[+] /usr (external python-3.11-ludrksmzgh4fondsvxz45dqyqsfc2b6i)
[+] /home/balay/.spack/bootstrap/store/linux-fedora37-x86_64/gcc-12.1.1/re2c-2.2-tamq2ghnyvbfalyvdrlvk3nu2pnk77lk
[+] /home/balay/.spack/bootstrap/store/linux-fedora37-x86_64/gcc-12.1.1/clingo-bootstrap-spack-zq6upukatixdbaveqeu23nfq5pkgfmbd
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Warning: Spec clingo-bootstrap@spack%gcc@12.1.1~docs~ipo+python~static_libstdcpp build_type=Release arch=linux-fedora37-x86_64 ^bison@3.8.2%gcc@12.1.1 arch=linux-fedora37-x86_64 ^cmake@3.24.1%gcc@12.1.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora37-x86_64 ^python@3.11%gcc@12.1.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,7d40923,f2fd060 arch=linux-fedora37-x86_64 ^re2c@2.2%gcc@12.1.1 arch=linux-fedora37-x86_64 did not provide module clingo
==> Error: cannot bootstrap the "clingo" Python module from spec "clingo-bootstrap@spack+python %gcc target=x86_64" due to the following failures:
github-actions-v0.2 raised ValueError: source is not trusted
github-actions-v0.1 raised ValueError: source is not trusted
Run `spack --debug ...` for more detailed errors
```

### Information on your system

```console
[balay@pj01 spack.y]$ ./bin/spack debug report
* **Spack:** 0.19.0.dev0 (8e5ccddc139b32a4a23f63c9dd631dabf3263cad)
* **Python:** 3.11.0rc1
* **Platform:** linux-fedora37-skylake
* **Concretizer:** clingo


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output
```

[debug.log](https://github.com/spack/spack/files/9509294/debug.log)

**Comment by balay** [comment]
I guess its similar to https://github.com/spack/spack/issues/29507

And same work-arounds work here..

```
[balay@pj01 spack.y]$ python3.9 ./bin/spack spec zlib
Input spec
--------------------------------
zlib

Concretized
--------------------------------
zlib@1.2.12%gcc@12.1.1+optimize+pic+shared patches=0d38234 arch=linux-fedora37-skylake

````
and:
```
python3.9 ./bin/spack install python@3.11.0rc1%gcc@12.1.1
/home/balay/spack.y/opt/spack/linux-fedora37-skylake/gcc-12.1.1/python-3.11.0rc1-aq466luj535dgn6zfrbtqsxiypzcklpl/bin/python ./bin/spack spec zlib
<so now>
[balay@pj01 spack.y]$ ./bin/spack spec zlib
Input spec
--------------------------------
zlib

Concretized
--------------------------------
zlib@1.2.12%gcc@12.1.1+optimize+pic+shared patches=0d38234 arch=linux-fedora37-skylake
```

**Comment by balay** [comment]
Add ref: https://github.com/spack/spack/pull/33505

**Comment by alalazo** [comment]
This should be fixed by #33665 Let me know if this is not the case

**Comment by balay** [comment]
Works now. Thanks!
